src/v/serde/serde.h

```diff
@@ -157,6 +159,20 @@ struct is_absl_node_hash_map<absl::node_hash_map<Args...>> : std::true_type {};
 template<typename T>
 inline constexpr bool is_absl_node_hash_map_v = is_absl_node_hash_map<T>::value;
 
+template<typename T>
+struct is_absl_btree_set : std::false_type {};
+template<typename... Args>
+struct is_absl_btree_set<absl::btree_set<Args...>> : std::true_type {};
+template<typename T>
+inline constexpr bool is_absl_btree_set_v = is_absl_btree_set<T>::value;
```

Comment:
question: what is the difference between this construct and something like `std::is_same_v<T, absl::btree_set>` ?

Reply:
You could probably make it directly a concept check:
```c++
template<class T>
concept is_absl_btree_set = requires(T x) { 
    { absl::btree_set{x} } -> std::same_as<T>;
};
```
Which is a bit cheating.